src/Components/Icon.tsx

```diff
@@ -26,11 +26,13 @@ export interface IconProps
   onClick?: () => void
 }
 
-const Icon: React.SFC<IconProps> = props => (
-  <div className={props.className} style={props.style} onClick={props.onClick}>
-    {icons[props.name]}
-  </div>
-)
+const Icon: React.SFC<IconProps> = ({
+  color,
+  font,
+  fontSize,
+  name,
+  ...props
+}) => <div {...props}>{icons[name]}</div>
 
 export default styled(Icon)`
```

Comment:
@damassi Do you know if we can type this? I guess the `.attrs<>({})` thing wouldn't work here. Maybe because we're calling this like a function it'd just be `styled<IconProps>(Icon)`?

Reply:
@alloy might have an idea but i was playing with `styled<IconProps>(Icon)` a few days ago and it didn't seem valid.

Reply:
Another option might be to switch it around? i.e. create the styled container first and export the icon component which uses that styled container. That way it’s easy to type with `React.SFC<…>`.

Reply:
This can be done separately from this PR imo.